ironfish/src/metrics/meter.ts

```diff
@@ -106,10 +115,10 @@ export class Meter {
     const elapsedMs = now - this._intervalLastMs
     const rateSec = elapsedMs === 0 ? 0 : (this._count / elapsedMs) * 1000
 
-    this._rate1s.add(rateSec)
-    this._rate5s.add(rateSec)
-    this._rate1m.add(rateSec)
-    this._rate5m.add(rateSec)
+    this._rate1s.add(rateSec, this.METER_TIME_INTERVALS_MS.rate1s / this._intervalMs)
+    this._rate5s.add(rateSec, this.METER_TIME_INTERVALS_MS.rate5s / this._intervalMs)
+    this._rate1m.add(rateSec, this.METER_TIME_INTERVALS_MS.rate1m / this._intervalMs)
+    this._rate5m.add(rateSec, this.METER_TIME_INTERVALS_MS.rate5m / this._intervalMs)
```

Comment:
It took me some time to read various sources, and a few implementations of EWMA but I'm not sure this part is correct. I looked in the source code of `ewma_bandwidth_estimator` where this code came from in the shaka library, and they use this a bit different to adjust the alpha for non evenly spaced points.

1. The half-life in the constructor should be in millisecond units. Shaka uses seconds because they keep all their data in seconds consistently. https://github.com/shaka-project/shaka-player/blob/0335b2af2efea6ceda83e536e12094e4cc942a25/lib/abr/ewma_bandwidth_estimator.js#L22-L34
2. Weight is used to adjust the alpha, from unevenly spaced data points which EWMA requires. Shaka just uses elapsed time only, regardless of the EWMA period / half life. https://github.com/shaka-project/shaka-player/blob/0335b2af2efea6ceda83e536e12094e4cc942a25/lib/abr/ewma_bandwidth_estimator.js#L97

Reply:
1 - Hmm, not sure why the half-life unit should be in ms. We're also delineating our metrics in s, like Shaka. We record values as `rateSec`. 
 
2 - Yea, Shaka uses different half-life periods but a consistent weight value when inserting data. I was doing it the other way around (same half-life, but weight adj). I've reversed this now.